File: ProductCrudKnockOut/Models/CompanyModel.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.Text.Json.Serialization;

namespace ProductCrudKnockOut.Models
{
    public class CompanyModel
    {
        public int Id { get; set; }

        public string CompanyName {  get; set; }

        public string ProductManufactured {  get; set; }

        public DateTime CompanyEstd { get; set; }

        public string CompanyPhNo { get; set;}

        public int ProductId {  get; set; }
       
        [ValidateNever]
        [JsonIgnore]
        public ProductModel ? Product {  get; set; }
    }
}

File: ProductCrudKnockOut/wwwroot/js/Company/CompanyModel.js
/// <reference path="knockout.js" />

var companyModel = function (item)
{
    var self = this;
    var item = item || {};
    self.id = item.id || 0;
    self.companyName = item.companyName || '';
    self.productId = item.productId || '';
    self.productManufactured = item.productManufactured || '';
    self.companyEstd = item.companyEstd || '';
    self.companyPhNo = item.companyPhNo || '';
}

var productVM = function (item)
{
    var self = this;
    var item = item || {};
    self.productId = item.productId || 0;
    self.productName = item.productName || '';
}